packages/core/components/ExternalInput/index.tsx

```diff
@@ -18,6 +18,10 @@ export const ExternalInput = ({
   const [data, setData] = useState<Record<string, any>[]>([]);
   const [isOpen, setOpen] = useState(false);
   const [selectedData, setSelectedData] = useState(value);
+  const keys = useMemo(
+    () => Object.keys(data).filter((key) => typeof data[key] === "string"),
+    [data]
+  );
```

Comment:
Without this fix, I presume that Puck just crashes?

If we're keeping this fix, I think we should probably also support other renderable types, like `number`.

You could always filter the data returned in your adaptor, rather than here, although I suppose that means Puck can still break if you forget to do that.

Reply:
Yeah, it just crashes.

Wouldn't filtering the data out in the adaptor just remove it completly? I encountered this crash when using external `_data` with an `items` property that I wanted to be spread into my component. I want to keep that data, I just want to not render it.

Perhaps the best solution is to stop it crashing now, then investigate more powerful solutions as part of the issue I've created?

Reply:
That's a good point - yes it would remove it completely!